platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/providers/FieldProvider.kt

```diff
@@ -115,30 +111,238 @@ internal fun ByteArray.toFieldValue() =
     }
 
 /**
- * Converts a [Duration] into a [FieldValue] using the specified [DurationUnit].
+ * Holds parallel arrays of field keys and values.
+ * This is an optimization to avoid creating wrapper objects when passing to JNI.
+ * keys[i] corresponds to values[i].
+ *
+ * Note to construct it please use, fieldsOf(vararg pairs: Pair<String, String>): Fields
+ *
+ * @property keys Array of field keys
+ * @property values Array of field values corresponding to keys
+ */
+@Suppress("UndocumentedPublicClass", "UndocumentedPublicFunction")
+class Fields internal constructor(
```

Comment:
if this class is indeed public (and you do provide kdocs above) why mark it as `"UndocumentedPublicClass", "UndocumentedPublicFunction"` ?